schema/test/unit/tables/revision_status_test.sql

```diff
@@ -0,0 +1,100 @@
+begin;
+
+select plan(10);
+
+select has_table('cif', 'revision_status', 'table cif.revision_status exists');
+
+select columns_are(
+  'cif',
+  'revision_status',
+  ARRAY[
+    'name',
+    'created_at',
+    'created_by',
+    'updated_at',
+    'updated_by',
+    'archived_at',
+    'archived_by',
+    'is_amendment_specific'
+  ],
+  'columns in cif.revision_status match expected columns'
+);
+
+-- Test setup
+set jwt.claims.sub to '11111111-1111-1111-1111-111111111111';
+
+-- cif_admin
+set role cif_admin;
+select concat('current user is: ', (select current_user));
+
+select lives_ok(
+  $$
+    select * from cif.revision_status
+  $$,
+    'cif_admin can view all data in revision_status table'
+);
+
+select throws_like(
+  $$
+    delete from cif.revision_status where name='changed by admin';
+  $$,
+  'permission denied%',
+    'Administrator cannot delete rows from table revision_status'
+);
+
+-- cif_internal
+set role cif_internal;
+select concat('current user is: ', (select current_user));
+
+select results_eq(
+  $$
+    select count(*) from cif.revision_status
+  $$,
+  ARRAY['5'::bigint], -- 5 because 'Approved' was removed
+    'cif_internal can view all data from revision_status table'
+);
+
+select throws_like(
+  $$
+    delete from cif.revision_status where name='changed by internal';
+  $$,
+  'permission denied%',
+    'cif_internal cannot delete rows from revision_status table'
+);
+
+select is (
+  (select count(*) from cif.revision_status where is_amendment_specific = false),
+  2::bigint,
+  'sets is_amendment_specific to false when status is Draft or Applied'
+);
+
+select is (
+  (select count(*) from cif.revision_status where is_amendment_specific = true),
+  3::bigint,
+  'sets is_amendment_specific to false when status is Discussion / Pending Province Approval / Pending Proponent Signature'
+);
+
+
+-- project_revision table changes
+
+select is (
+  (select count(*) from cif.project_revision where revision_status = 'Approved'),
+  0::bigint,
+  'replaces Approved status with Applied status in project_revision table'
+);
+
+select throws_ok(
+  $$
+    insert into cif.project_revision(project_id, revision_status)
+values
+  (7,null);
+
+  $$,
+  'null value in column "revision_status" of relation "project_revision" violates not-null constraint',
+  'revision_status cannot be null'
+);
```

Comment:
We could use `col_not_null(...)` from [pgTap](https://pgtap.org/documentation.html#col_not_null) instead